packages/firestore/src/api/credentials.ts

```diff
@@ -395,17 +398,46 @@ interface Gapi {
 export class FirstPartyToken implements Token {
   type = 'FirstParty' as TokenType;
   user = User.FIRST_PARTY;
-  headers = new Map();
+  private _headers = new Map();
+
+  constructor(
+    private readonly gapi: Gapi,
```

Comment:
Could this potentially be nullable?  I think part of the intent here is to reduce the dep on `gapi.js` for first party usage.

(I recognize that making this type nullable may cause a slight ripple effect in this file).

Reply:
I made it nullable, but I don't think it is of much difference..unless I misunderstood you.